src/Serializer/AbstractItemNormalizer.php

```diff
@@ -523,6 +523,9 @@ protected function getAttributeValue($object, $attribute, $format = null, array
             $this->resourceClassResolver->isResourceClass($className)
         ) {
             if (!is_iterable($attributeValue)) {
+                if ($type->isNullable() && null === $attributeValue) {
```

Comment:
```suggestion
                if (null === $attributeValue && $type->isNullable()) {
```

(the comparison costs less than the function call)